### PR TITLE
Added automatic version fetching for ubuntu

### PR DESCRIPTION
**Pull request by Jai-JAP**
- Fetch ubuntu versions directly from https://api.launchpad.net/devel/ubuntu/series (desktop flavors) and http://releases.ubuntu.com/streams/v1/com.ubuntu.releases:ubuntu-server.json (Server)
 - Remove obsolete releases {daily-canary, jammy-daily}
 - Format documents with shfmt

### CODE REVIEW

**Comment by philclifford** [comment]
Excellent!

One tweak though - jammy-daily is not quite obsolete :

https://cdimage.ubuntu.com/jammy/daily-live/current/ 
https://iso.qa.ubuntu.com/qatracker/milestones/429/builds

**Comment by Jai-JAP** [comment]
> Excellent!
> 
> One tweak though - jammy-daily is not quite obsolete :
> 
> https://cdimage.ubuntu.com/jammy/daily-live/current/ 
> https://iso.qa.ubuntu.com/qatracker/milestones/429/builds
> 

I'll fix that soon.

**Comment by Jai-JAP** [comment]
jammy-daily should be back working now

**Comment by philclifford** [comment]
Thanks

**Review by philclifford — APPROVED** [review]

**Review by flexiondotorg — CHANGES_REQUESTED** [review]
I can not accept pull requests that change the indentation project wide.

**Comment by Jai-JAP** [comment]
> I can not accept pull requests that change the indentation project wide.

I will fix that soon.

**Comment by flexiondotorg** [comment]
This patch is massive:

image

Please refactor this pull request to only provide the changes to add automatic version fetching for Ubuntu and **remove all whitespace-related changes**.

**Comment by Jai-JAP** [comment]
@flexiondotorg ig the changes you requested are resolved.

**Review by philclifford — APPROVED** [review]

**Review by philclifford — CHANGES_REQUESTED** [review]
The purely-whitespace changes would be better dealt with in another (possibly existing) PR. (But at least now they are restricted to a couple of functions. - this is not a show-stopper for me but ...)
 
The results do not correctly align with the available releases so there is either a flaw in the logic or in what the api presents.
Some examples below (`./quickget` is the PR version) showing missing 23.10 interims , missing `jammy-daily` options and non-existent early releases.

```shell
$
$quickget ubuntu-unity
ERROR! You must specify a release.
 - Releases: 23.04 23.10 daily-live daily-canary
$./quickget ubuntu-unity
ERROR! You must specify a release.
 - Releases: 23.04 daily-live
$quickget xubuntu
ERROR! You must specify a release.
 - Releases: 16.04 18.04 20.04 22.04 23.04 23.10 jammy-daily daily-live daily-canary daily-legacy                                                                        
$./quickget xubuntu
ERROR! You must specify a release.
 - Releases: 14.04 16.04 18.04 20.04 22.04 23.04 23.10 daily-live daily-legacy jammy-daily                                
$quickget ubuntucinnamon
ERROR! You must specify a release.
 - Releases: 23.04 23.10 daily-live daily-canary
$./quickget ubuntucinnamon
ERROR! You must specify a release.
 - Releases: 23.04 daily-live
$quickget ubuntu-server
ERROR! You must specify a release.
 - Releases: 18.04 20.04 22.04 23.04 23.10 daily-live
$./quickget ubuntu-server
ERROR! You must specify a release.
 - Releases: 18.04 20.04 22.04 23.04 23.10 daily-live
$quickget ubuntu
ERROR! You must specify a release.
 - Releases: 14.04 16.04 18.04 20.04 22.04 23.04 23.10 jammy-daily daily-live daily-canary daily-legacy    
$./quickget ubuntu
ERROR! You must specify a release.
 - Releases: 14.04 16.04 18.04 20.04 22.04 23.04 23.10 daily-live daily-legacy                                                                     
$quickget ubuntu-mate
ERROR! You must specify a release.
 - Releases: 16.04 18.04 20.04 22.04 23.04 23.10 jammy-daily daily-live daily-canary daily-legacy                                                                        
$./quickget ubuntu-mate
ERROR! You must specify a release.
 - Releases: 14.04 16.04 18.04 20.04 22.04 23.04 23.10 daily-live daily-legacy jammy-daily         
```

File: quickget [hunk 1]
```diff
@@ -1659,13 +1638,13 @@ function get_manjaro() {
     [[ ${RELEASE} != "sway" ]] && MANIFEST="$(wget -qO- https://gitlab.manjaro.org/web/iso-info/-/raw/master/file-info.json)"
 
     [[ ${EDITION} == "minimal" && ${TYPE} != "sway" ]] && EDITION=".minimal" || EDITION=""
-    
-    if [[ ${RELEASE} != "sway" ]]; then 
+
```
[the remark this answers is not in the document]
Comment (Jai-JAP) [reply quickget:1641]: > Should not be changing vanillaos or other non-ubuntu functions in this PR

I didn't make any manual changes here. Maybe editorconfig automatically removing trailing whitespaces.

**Comment by Jai-JAP** [comment]
@philclifford the version of quickget u are using is not from this PR as it removed all traces of daily-canary which is present in your output

**Comment by Jai-JAP** [comment]
@philclifford Everything should be working fine now. Can you test again?

**Comment by Jai-JAP** [comment]
@philclifford Edubuntu is fixed now.

**Pull request merged**